src/service/navigation.js

```diff
@@ -553,61 +554,62 @@ export class Navigation {
         }
       }
 
-      // Safari 13: Temporarily remove viewer query params from iframe
-      // (e.g. amp_js_v, usqp) to prevent document.referrer from being reduced
-      // to eTLD+1 (e.g. ampproject.org).
+      // ITP 2.3 mitigation. See https://github.com/ampproject/amphtml/issues/25179.
       const {win} = this.ampdoc;
       const platform = Services.platformFor(win);
       const viewer = Services.viewerForDoc(element);
       if (
-        isExperimentOn(
-          this.ampdoc.win,
-          'remove-viewer-query-params-on-navigate'
-        ) &&
         fromLocation.search &&
         platform.isSafari() &&
         platform.getMajorVersion() >= 13 &&
         viewer.isProxyOrigin() &&
         viewer.isEmbedded()
       ) {
-        dev().info(
-          TAG,
-          'Removing iframe query string before navigation:',
-          fromLocation.search
-        );
-        const original = fromLocation.href;
-        const noQuery = `${fromLocation.origin}${fromLocation.pathname}${fromLocation.hash}`;
-        win.history.replaceState(null, '', noQuery);
-
-        const restoreQuery = () => {
-          const currentHref = win.location.href;
-          if (currentHref == noQuery) {
-            dev().info(TAG, 'Restored iframe URL with query string:', original);
-            win.history.replaceState(null, '', original);
-          } else {
-            dev().error(
-              TAG,
-              'Unexpected iframe URL change:',
-              currentHref,
-              noQuery
-            );
-          }
-        };
-
-        // For blank_, restore query params after the new page opens.
-        if (target === '_blank') {
-          win.setTimeout(restoreQuery, 0);
-        } else {
-          // For _top etc., wait until page is restored from page cache (bfcache).
-          // https://webkit.org/blog/516/webkit-page-cache-ii-the-unload-event/
-          win.addEventListener('pageshow', function onPageShow(e) {
-            if (e.persisted) {
-              restoreQuery();
-              win.removeEventListener('pageshow', onPageShow);
-            }
-          });
-        }
+        this.removeViewerQueryBeforeNavigation_(win, fromLocation, target);
+      }
+    }
+  }
+
+  /**
+   * Temporarily remove viewer query params from iframe (e.g. amp_js_v, usqp)
+   * to prevent document.referrer from being reduced to eTLD+1 (e.g. ampproject.org).
```

Comment:
What would be an example previous document.referrer value? `example.ampproject.org --> ampproject.org`?

Reply:
Actually the full URL e.g. `example-com.ampproject.org/v/s/example.com/articles/foo?bar=baz`. 

With this fix we'll lose `?bar=baz` but keep the full hostname and pathname.